src/faim_hcs/MetaSeriesUtils.py

```diff
@@ -152,15 +154,45 @@ def sort_key(d):
 
         img[pos_y : pos_y + d[0].shape[0], pos_x : pos_x + d[0].shape[1]] = d[0]
 
-    return img
+        # Create the FOV ROI table for the site in physical units
+        fov_rois.append(
+            (
+                _stage_label(d[1]),
+                pos_y * d[1]["spatial-calibration-y"],
+                pos_x * d[1]["spatial-calibration-x"],
+                0.0,  # Hard-coded z starting position
+                d[0].shape[0] * d[1]["spatial-calibration-y"],
+                d[0].shape[1] * d[1]["spatial-calibration-x"],
+                1.0,  # Hard-coded z length (for 2D planes), to be overwritten if
+                # the 2D planes are assembled into a 3D stack
+            )
+        )
+
+    roi_tables = create_ROI_tables(fov_rois, shape, calibration_dict=d[1])
+
+    return img, roi_tables
 
 
 def _pixel_pos(dim: str, data: dict):
     return np.round(data[f"stage-position-{dim}"] / data[f"spatial-calibration-{dim}"])
 
 
+def _stage_label(data: dict):
+    """Get the FOV string for a given FOV dict"""
```

Comment:
I suggest being explicit:

```suggestion
    """Get the field of view (FOV) string for a given FOV dict"""
```